hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/impl/TestVectoredReadUtils.java

```diff
@@ -254,6 +254,20 @@ public void testSortAndMerge() {
     assertOrderedDisjoint(list4, 16, 700);
   }
 
+  @Test
+  public void testMergeSortedRanges() {
+    List<FileRange> input = asList(
+        createFileRange(13816220, 24, null),
+        createFileRange(13816244, 7423960, null)
+    );
+    assertIsNotOrderedDisjoint(input, 100, 800);
+    final List<CombinedFileRange> outputList = mergeSortedRanges(
+        sortRangeList(input), 100, 1001, 2500);
```

Comment:
the maxSize passed to mergeSortedRanges is 2500, and merged length would be 7424100. So, this test is supposed to fail.